### PR TITLE
Allow for building ResettingHistogram metrics using SfxMetrics

### PR DESCRIPTION
**Pull request by udaysagar2177**
This allows attaching extra dimensions and easy replacement
of sfxMetrics.histogram(..) calls.

### CODE REVIEW

File: signalfx-codahale/src/main/java/com/signalfx/codahale/SfxMetrics.java [hunk 1]
```diff
@@ -186,6 +188,24 @@ public Histogram histogram(String metricName, Map<String, String> dimensions) {
...
+     * clears the resorvoir after each {@link Reservoir#getSnapshot()}. Clearing resorvoir after
+     * snapshot removes the effect of old data points and provides better visibility (where needed).
+     *
+     * @param metricName
+     *         The metric name.
+     * @param dimensions
+     *         Additional dimension key/value pairs, as a map.
+     * @return The {@link Histogram} instance.
+     */
+    public Histogram resettingHistogram(String metricName, Map<String, String> dimensions) {
```
Comment (mpetazzoni) [inline signalfx-codahale/src/main/java/com/signalfx/codahale/SfxMetrics.java]: Can you also provide the variant with `String... dimensions` ?

**Comment by udaysagar2177** [comment]
I just found SignalfxReporterTest, let me add some tests.

**Comment by udaysagar2177** [comment]
After https://github.com/signalfx/signalfx-java/pull/75 is merged, I will rebase this change on top of master to remove "cleanup SignalFxReporterTest" commit from this PR.

File: signalfx-codahale/src/main/java/com/signalfx/codahale/SfxMetrics.java (hunk 1 above)
Comment (udaysagar2177) [reply signalfx-codahale/src/main/java/com/signalfx/codahale/SfxMetrics.java]: Done

**Pull request merged**